common/src/main/java/net/whimxiqal/journey/data/DataManagerImpl.java

```diff
@@ -41,80 +31,74 @@
-            addBatchesToStatement("/data/sql/schema/sqlite.sql", statement);
-            statement.executeBatch();
-          } catch (SQLException e) {
-            setupSchema = false;
-            e.printStackTrace();
-          }
-        }
+
+        if (currentVersionNumber < DataVersion.latest().internalVersion) {
+          databaseVersion = sqliteDataVersionHandler.runMigrations(currentVersion);
+        } else databaseVersion = DataVersion.latest();
+        break;
+      case MYSQL:
+        MySqlConnectionController mysqlController = new MySqlConnectionController();
+
+        MysqlDataVersionHandler mysqlDataVersionHandler = new MysqlDataVersionHandler(mysqlController);
+        currentVersion = mysqlDataVersionHandler.version();
+        currentVersionNumber = currentVersion.internalVersion;
+
+        personalWaypointManager = new SqlPersonalWaypointManager(mysqlController);
+        publicWaypointManager = new SqlPublicWaypointManager(mysqlController);
+        pathRecordManager = new SqlPathRecordManager(mysqlController);
+        tunnelDataManager = new SqlTunnelDataManager(mysqlController);
+
+        if (currentVersionNumber < DataVersion.latest().internalVersion) {
+          databaseVersion = mysqlDataVersionHandler.runMigrations(currentVersion);
+        } else databaseVersion = DataVersion.latest();
         break;
-//      case MYSQL:
-//        MySqlConnectionController mysqlController = new MySqlConnectionController();
-//        personalWaypointManager = new SqlPersonalWaypointManager(mysqlController);
-//        publicWaypointManager = new SqlPublicWaypointManager(mysqlController);
-//        pathRecordManager = new SqlPathRecordManager(mysqlController);
-//        tunnelDataManager = new SqlTunnelDataManager(mysqlController);
-//        if (setupSchema) {
-//          try (Connection connection = mysqlController.establishConnection()) {
-//            Statement statement = connection.createStatement();
-//            addBatchesToStatement("/data/sql/schema/mysql.sql", statement);
-//            statement.executeBatch();
-//          } catch (SQLException e) {
-//            setupSchema = false;
-//            e.printStackTrace();
-//          }
-//        }
-//        break;
       default:
         throw new RuntimeException();
     }
-    if (setupSchema) {
-      DataVersion.writeVersion(DataVersion.latest());
-    }
-  }
-
-  private void addBatchesToStatement(String queryResource, Statement statement) throws SQLException {
-    InputStream resourceStream = this.getClass().getResourceAsStream(queryResource);
-    if (resourceStream == null) {
-      throw new NoSuchElementException("Cannot find resource at path " + queryResource);
-    }
-    for (String query : new BufferedReader(new InputStreamReader(resourceStream))
-        .lines().collect(Collectors.joining("\n")).split(";")) {
-      statement.addBatch(query);
-    }
   }
 
   @Override
   public DataVersion version() {
```

Comment:
This method is called on plugin initialization, so it would be nice to add there that if a database version mis-match is identified, the plugin is disabled with Bukkit, since the plugin should be deemed effectively unusable.